src/components/common/CarouselComponent.jsx

```diff
@@ -0,0 +1,127 @@
+import { useRef, useState, useEffect, Children, cloneElement } from 'react';
+import styled from 'styled-components';
+
+const CarouselContainer = styled.div`
+  width: 100%;
+  overflow-x: auto;
+  overflow-y: hidden;
+  display: flex;
+  gap: ${({ $gap }) => $gap};
+
+  /* 스크롤바 숨김 */
+  scrollbar-width: none;
+  -ms-overflow-style: none;
+  &::-webkit-scrollbar {
+    display: none;
+  }
+
+  /* 스크롤 스냅 */
+  scroll-snap-type: x mandatory;
+  scroll-behavior: smooth;
+  -webkit-overflow-scrolling: touch;
+
+  cursor: grab;
+  &:active {
+    cursor: grabbing;
+  }
+  -webkit-user-select: none;
+  -moz-user-select: none;
+  user-select: none;
+
+  touch-action: pan-y;
+
+  > * {
+    flex: 0 0 100%; // 각 아이템이 컨테이너 너비를 100% 차지하도록 수정
+    scroll-snap-align: start;
+    scroll-snap-stop: always;
+  }
+`;
+
+const Carousel = ({ children, gap }) => {
```

Comment:
대신, 여기에 주석을 남겨 Carousel 컴포넌트를 사용하려는 사람이 flex를 어떻게 지정해야 하는지 알려주세요. [JSDoc](https://jsdoc.app/about-getting-started)을 활용하면 VSCode에서 Carousel 글씨에 마우스를 올렸을 때 주석을 보여줘요. 그리고 PR에도 이 컴포넌트를 어떻게 사용해야 하는지 설명을 남기면 아주 좋아요!
```suggestion
/**
 * Carousel의 직계 자식 컴포넌트에 flex를 지정해 주세요.
 * 한 슬라이드에 하나의 요소만 보여주고 싶다면 `flex: 0 0 100%`을 지정해 주세요.
 * 여러 요소를 동시에 보여주고 싶다면 `flex: 0 0 아이템너비`를 지정해 주세요.
 * @param gap - 직계 자식 컴포넌트 간 간격을 지정해요.
 * @param setIndex - 부모 컴포넌트에 index state를 만들어 setter를 넘겨 주세요.
 */
const Carousel = ({ children, gap }) => {
```